draft-ietf-tls-esni.md

```diff
@@ -430,6 +430,37 @@ the ClientHelloOuter. The server then computes the digest of the reconstructed
 ClientHelloInner. If the digest does not equal OuterExtensions.inner_digest,
 then the server MUST abort the connection with a "decrypt_error" alert.
 
+# The "handshake_padding" Extension {#handshake-padding}
+
+Encrypting the ClientHelloInner may not be sufficient to protect sensitive
+parameters. In particular, the length of the ClientHelloInner, or the length of
+the backend server's response, may leak information about these parameters. The
+client and backend server can use the "handshake_padding" extension in order to
+mitigate this leakage.
+
+A new extension type is defined, which the client MAY include in its
+ClientHello. It MUST NOT offer this extension unless it also advertises support
+for TLS 1.3 or above.
+
+~~~
+    enum {
+       handshake_padding(0xff05), (65535)
+    } ExtensionType;
+~~~
+
+When solicited by the client, the server MUST ignore the "handshake_padding"
+extension if it negotiates TLS 1.2 or below. Otherwise, it MAY reply by
+including the extension in its EncryptedExtensions message.
+
+The payload of the "handshake_padding" extension is any number of zeros bytes.
+The extension's recipient MAY verify the payload is correct and abort with an
+"illegal_parameter" alert if not.
+
+The number of zero bytes is referred to as the padding length. Note that unlike
+{{!RFC7685}}, the padding length is not encoded by the extension payload itself.
```

Comment:
Nit: I think this should be `{{?RFC7685}}` to mark it as an informational reference rather than a normative reference.